tests/test_enterprise/views/test_enterprise_proxy_login_view.py

```diff
@@ -82,24 +82,33 @@ def test_redirect_no_tpa(self, use_next):
         expected_url = LMS_LOGIN_URL + '?' + query_params.urlencode()
         self.assertRedirects(response, expected_url, fetch_redirect_response=False)
 
-    @ddt.data(
-        {'use_next': True},  # 'next' query param is passed in URL
-        {'use_next': False},  # 'next' query param not passed, default to LP URL with slug
-    )
-    @ddt.unpack
-    def test_tpa_redirect(self, use_next):
+    def test_tpa_redirect_no_next_param(self):
         """
-        Verify the view adds the next param and the tpa_hint to the redirect if the enterprise has an identity provider.
+        When 'next' url param is absent:
+          The proxy login redirects to default learner portal url,
+          url should contain: ?next=<learner portal url>&<tpa_hint=customer idp>
         """
-        url = self._get_url_with_params(use_next=use_next)
+        url = self._get_url_with_params(use_next=False)
         response = self.client.get(url)
         query_params = QueryDict(mutable=True)
-        if use_next:
-            next_url = self.next_url
-        else:
-            learner_portal_url = settings.ENTERPRISE_LEARNER_PORTAL_BASE_URL
-            next_url = learner_portal_url + '/' + self.enterprise_customer.slug
-        query_params['next'] = f'{next_url}?tpa_hint={self.enterprise_customer.identity_provider}'
+        learner_portal_url = settings.ENTERPRISE_LEARNER_PORTAL_BASE_URL
+        next_url = learner_portal_url + '/' + self.enterprise_customer.slug
+        query_params['next'] = next_url
+        query_params['tpa_hint'] = self.enterprise_customer.identity_provider
+        expected_url = LMS_LOGIN_URL + '?' + query_params.urlencode()
+        self.assertRedirects(response, expected_url, fetch_redirect_response=False)
+
+    def test_tpa_redirect_with_next_param(self):
```

Comment:
I felt like having conditionals in the test for logic branching is not a good idea, hence I split these two cases out

Reply:
The conditional in conjunction with DDT was to avoid having this duplicate logic in tests. I don't agree with this change but this also isn't a hill I'm willing to die on so it's up to you

Reply:
I hear you on the duplication. It is a fine balance sometimes between choosing duplication vs choosing conditional logic - esp in tests. Typically I choose to avoid the latter and also we can address duplication such as this (if more than two tests for example), by util methods and such... I decided not to do that here yet but yeah if we do more of these, worth addressing the dup too. Also, sorry just me, I just have not yet gained a liking for ddt (parameterized testing in general) unless a large no. of data driven scenarios are being handled.